app/components/post_list/post/burn_on_read/unrevealed/index.tsx

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback} from 'react';
+
+import {deletePost, revealBoRPost} from '@actions/remote/post';
+import Button from '@components/button';
+import {useServerUrl} from '@context/server';
+import {useTheme} from '@context/theme';
+import {PostModel} from '@database/models/server';
+import {getFullErrorMessage, getServerError} from '@utils/errors';
+import {showBoRPostExpiredSnackbar} from '@utils/snack_bar';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+
+import {BOR_GLOBALLY_EXPIRED_POST_ERROR_CODE, BOR_POST_EXPIRED_FOR_USER_ERROR_CODE} from './constants';
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => ({
+    buttonBackgroundStyle: {
+        backgroundColor: changeOpacity(theme.centerChannelColor, 0.08),
+        height: 56,
+        marginBottom: 8,
+    },
+    buttonTextStyle: {
+        color: changeOpacity(theme.centerChannelColor, 0.56),
+    },
+}));
+
+type Props = {
+    post: PostModel;
+}
+
+export default function UnrevealedBurnOnReadPost({post}: Props) {
+    const theme = useTheme();
+    const styles = getStyleSheet(theme);
+    const serverUrl = useServerUrl();
+
+    const handleRevealPost = useCallback(async () => {
+        const {error} = await revealBoRPost(serverUrl, post.id);
+        if (error) {
+            showBoRPostExpiredSnackbar(getFullErrorMessage(error));
+
+            const serverError = getServerError(error);
+            if (serverError === BOR_POST_EXPIRED_FOR_USER_ERROR_CODE || serverError === BOR_GLOBALLY_EXPIRED_POST_ERROR_CODE) {
+                deletePost(serverUrl, post);
+            }
+        }
+
+    }, [serverUrl, post]);
+
+    return (
+        <Button
+            text={'View message'}
+            iconName='eye-outline'
+            theme={theme}
+            backgroundStyle={styles.buttonBackgroundStyle}
+            textStyle={styles.buttonTextStyle}
```

Comment:
Let's avoid passing styles, and see if you can get the same with the `emphasis` and `size` props.

Reply:
Any comment on this?